nni/tools/nnictl/algo_management.py

```diff
@@ -35,11 +35,18 @@ def _do_verify_import(fullName):
 def algo_reg(args):
     meta_list = read_reg_meta_list(args.meta_path)
     for meta in meta_list:
-        if get_registered_algo_meta(meta['builtinName']) is not None:
-            print_error('builtinName {} already registered'.format(meta['builtinName']))
-            return
-        verify_algo_import(meta)
-        save_algo_meta_data(meta)
+        old = get_registered_algo_meta(meta['builtinName'])
+        if old is None:
+            verify_algo_import(meta)
+            save_algo_meta_data(meta)
+        elif old['source'] != 'nni':
+            verify_algo_import(meta)
+            print_green(f'Updating exist algorithm')
+            remove_algo_meta_data(meta['builtinName'])
+            save_algo_meta_data(meta)
+        else:
+            print_error(f'Cannot overwrite builtin algorithm')
+            raise
```

Comment:
If `print_error()` function contains `raise`, then does not need to add `raise` again. BTW, does `raise` print process stack message in screen? Suggest to use `exit(1)` for clean message.